src/lib/CMakeLists.txt

```diff
@@ -222,9 +222,8 @@ set(
     operators/table_scan.cpp
     operators/table_scan.hpp
     operators/table_scan/abstract_table_scan_impl.hpp
-    operators/table_scan/base_single_column_table_scan_impl.cpp
-    operators/table_scan/base_single_column_table_scan_impl.hpp
-    operators/table_scan/base_table_scan_impl.hpp
+    operators/table_scan/abstract_single_column_table_scan_impl.cpp
+    operators/table_scan/abstract_single_column_table_scan_impl.hpp
```

Comment:
I thought about this some more yesterday and don't have a drop-solution, nothing we can apply in this PR at least, so it's probably not worth discussing it here. 

But we agree that we loose performance here? Even in the case of scanning on a ReferenceSegment that references only one Chunk, we write the wrong ChunkID into the `matches` PosList...

Reply:
> But we agree that we loose performance here?

Not sure if a solution of this problem would be without performance implications, but yes, this might be an issue.

Reply:
I'll keep it in mind, maybe some idea pops up at some point